File: 3797-design-spreadsheet/3797-design-spreadsheet.java
class Spreadsheet {
    private Map<String,Integer> hash;
    private int r;
    public Spreadsheet(int rows) {
        this.hash = new HashMap<>();
        this.r = rows; 
    }
    
    public void setCell(String cell, int value) {
        hash.put(cell,value);
    }
    
    public void resetCell(String cell) {
        hash.put(cell,0);
    }
    
    public int getValue(String formula) {
        String x = formula.substring(1);
        String[] arr = x.split("\\+");
        String a = arr[0]; String b = arr[1];
        int ans = 0;
        if(a.matches("\\d+")) ans+=Integer.parseInt(a);
        if(b.matches("\\d+")) ans+=Integer.parseInt(b);
        if(hash.containsKey(a)) ans+=hash.get(a);
        if(hash.containsKey(b)) ans+=hash.get(b);
        return ans;
    }
}

/**
 * Your Spreadsheet object will be instantiated and called as such:
 * Spreadsheet obj = new Spreadsheet(rows);
 * obj.setCell(cell,value);
 * obj.resetCell(cell);
 * int param_3 = obj.getValue(formula);
 */

File: 3797-design-spreadsheet/3797-design-spreadsheet.py
class Spreadsheet:

    def __init__(self, rows: int):
        self.hash = defaultdict(int)
        self.r = rows
    def setCell(self, cell: str, value: int) -> None:
        self.hash[cell] = value
    def resetCell(self, cell: str) -> None:
        self.hash [cell] = 0
    def getValue(self, formula: str) -> int:
        x = formula[1:]
        arr = x.split('+')
        a,b = arr[0],arr[1]
        ans = 0
        if a.isdigit():
            ans+=int(a)
        if b.isdigit():
            ans+=int(b)
        if a in self.hash:
            ans+=self.hash[a]
        if b in self.hash:
            ans+=self.hash[b]
        return ans


# Your Spreadsheet object will be instantiated and called as such:
# obj = Spreadsheet(rows)
# obj.setCell(cell,value)
# obj.resetCell(cell)
# param_3 = obj.getValue(formula)